portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,19 +14,54 @@
 
 package com.google.sps.servlets;
 
+import com.google.gson.Gson;
+import java.util.ArrayList;
+import java.util.List;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
+
+
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  private ArrayList<String> quotes;
+
+  @Override
+    public void init() {
+        
+     quotes = new ArrayList<>();
+    quotes.add("A ship in port is safe, but that is not what ships are for - Grace Hopper");
+    quotes.add("They told me computers could only do arithmetic. - Grace Hopper");
+    quotes.add("A ship in port is safe, but that is not what ships are built for. - Grace Hopper");
+    }
+
+ private String convertToJson(ArrayList<String> data) {
+    String json ="";
+     json += "{\"comments\":[ ";
+    for (int i = 0 ; i < data.size(); i++){
+         json += "{";
+        json += "\"comment\": ";
+        json += "\"" + data.get(i)+ "\"";
```

Comment:
Looks good! No need to change anything, but consider what would happen if one of the quotes contained a double-quote.

`quotes.add("A wise man once said, \"Hi!\". -- Anonymous");`

The tutorial mentioned Json-encoding/decoding libraries like GSon that will handle this situation without extra work or worries. In the future you should avoid writing JSON by hand like this -- it's only for learning :)